packages/core/src/login.ts

```diff
@@ -0,0 +1,118 @@
+import fetch from 'node-fetch'
+import { exec } from 'child_process'
+import * as jose from 'jose'
+import { localFs } from './store'
+import { Logger } from './log'
+import { withSpinner } from './spinner'
+
+export class TokenExpiredError extends Error {
+  constructor() {
+    super('Token is expired')
+  }
+}
+
+const PERSISTENT_TOKEN_FILE_NAME = 'lc-access-token.json'
+
+export type TokesFileSchema = {
+  access_token: string
+  id_token: string
+}
+
+const deviceFlow = async (dataDir: string, loginUrl: string, logger: Logger) => {
+  const fs = localFs(dataDir)
```

Comment:
can probably avoid the localFs abstraction

Reply:
Left it there because it handled the error nicely